src/vm/gcenv.os.cpp

```diff
@@ -205,6 +205,31 @@ bool GCToOSInterface::VirtualRelease(void* address, size_t size)
     return !!::ClrVirtualFree(address, 0, MEM_RELEASE);
 }
 
+// Commit virtual memory range.
+// Parameters:
+//  size      - size of the virtual memory range
+//  alignment - requested memory alignment, 0 means no specific alignment requested
+//  flags     - flags to control special settings like write watching
+//  node      - NUMA node
+// Return:
+//  Starting virtual address of the committed range
+void* GCToOSInterface::VirtualCommitLargePages(size_t size, size_t alignment, uint32_t node)
+{
+    LIMITED_METHOD_CONTRACT;
+
+    void* pRetVal = nullptr;
+    if (node == NUMA_NODE_UNDEFINED)
+    {
+        pRetVal = ::ClrVirtualAlloc(nullptr, size, MEM_RESERVE | MEM_COMMIT | MEM_LARGE_PAGES, PAGE_READWRITE);
```

Comment:
This code is used on both Unix and Windows for the GC integrated in coreclr. However, the VirtualAlloc implementation in PAL doesn't support the MEM_LARGE_PAGES option, so it would not try to allocate large pages on Unix.
 We will need to add support for this flag to the PAL implementation too.

Reply:
@janvorli I did a quick glance and it seems that I'll need to plumb a few params but it shouldn't be too bad. To unblock the linux builds I'll define MEM_LARGE_PAGES but still not do huge tlb for the moment. Ironing out Windows issues. So maybe the first PR could be enabling it for Windows and the Linux, and maybe then MacOSX? MacOSX has `HAVE_VM_ALLOCATE`?

Reply:
That makes sense. And yes, macOS is the only OS that has the HAVE_VM_ALLOCATE defined.